components/studio/bin/hab-studio.sh

```diff
@@ -991,6 +991,11 @@ chroot_env() {
   if [ -n "${HAB_UPDATE_STRATEGY_FREQUENCY_MS:-}" ]; then
     env="$env HAB_UPDATE_STRATEGY_FREQUENCY_MS=$HAB_UPDATE_STRATEGY_FREQUENCY_MS"
   fi
+  # If a Habitat studio binary is set, then propagate it into the Studio's
+  # environment.
+  if [ -n "${HAB_STUDIO_BINARY:-}" ]; then
```

Comment:
How will this behave if `HAB_STUDIO_BINARY` is given as an absolute path (i.e., a binary that exists at a path on the host system that's not at that location inside the studio)? 

Is that a valid concern? I think I've lost track at which Inception level we may be on at this point in the code :smile:

Reply:
I've actually only tested it with `HAB_STUDIO_BINARY` set to an absolute path:
```
11:13:10 AM jbauman@ubuntu:~/some-non-hab-source-dir
➤ env HAB_STUDIO_BINARY=$HOME/habitat/components/studio/bin/hab-studio.sh hab studio enter
```
But it does seem to work if the binary is not at a location inside the studio as above.

Reply:
Oh, that makes total sense then; that binary determines what studio is run, so it's by definition relative to the workstation, and not the inside of the studio. 

I was one level of Inception too deep 😆